transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/ws/connector/ControlIT.java

```diff
@@ -126,21 +109,13 @@ public Object invoke(Invocation invocation) throws Throwable {
         "server.send.close.payload.length.126/handshake.response.and.frame" })
     public void shouldFailWebSocketConnectionWhenServerSendCloseFrameWithPayloadTooLong() throws Exception {
         final IoHandler handler = context.mock(IoHandler.class);
-        final CountDownLatch latch = new CountDownLatch(1);
 
         context.checking(new Expectations() {
             {
                 oneOf(handler).sessionCreated(with(any(IoSessionEx.class)));
                 oneOf(handler).sessionOpened(with(any(IoSessionEx.class)));
-                oneOf(handler).exceptionCaught(with(any(IoSessionEx.class)), with(any(Throwable.class)));
-                oneOf(handler).sessionClosed(with(any(IoSessionEx.class)));
-                will(new CustomAction("Latch countdown") {
-                    @Override
-                    public Object invoke(Invocation invocation) throws Throwable {
-                        latch.countDown();
-                        return null;
-                    }
-                });
+                allowing(handler).exceptionCaught(with(any(IoSessionEx.class)), with(any(Exception.class)));
+                allowing(handler).sessionClosed(with(any(IoSessionEx.class)));
```

Comment:
In cases where the client sends a WsClose frame and the server does not send a WsClose response the client (WsnConnector) goes into a five second wait before abruptly closing the connection and causing exceptionCaught and session closed to be fired. Without the extra expectations the test would fail in this case. If you comment out those expectations and set up debug breakpoint on k3po.finish(), and wait there for 5 seconds before continuing, you will see the test failure. This means that we have to have these expectations to provide spurious test failures in cases where available CPU can fluctuate (i.e. Travis).